### PR TITLE
Add base-path parameter to the jenkins integ test

### PR DESCRIPTION
**Pull request by zelinh**
### Description
Add base-path parameter to the jenkins integ test.

I will have another following PR to our `opensearch-build` repo.

### Issues Resolved
Part of https://github.com/opensearch-project/opensearch-build/issues/3499

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by gaiksaya** [comment]
Looks like test cases are failing. Please ensure that you get the base path you are looking for using assertion. 
Thanks!

**Comment by zelinh** [comment]
> Looks like test cases are failing. Please ensure that you get the base path you are looking for using assertion. Thanks!

@gaiksaya Tests have been updated. Please help review here again. Thanks.

File: tests/jenkins/lib-testers/RunIntegTestScriptLibTester.groovy [hunk 1]
```diff
@@ -30,6 +30,8 @@ class RunIntegTestScriptLibTester extends LibFunctionTester {
 
     void configure(helper, binding) {
         binding.setVariable('env', ['BUILD_NUMBER': '987'])
+        binding.setVariable('env', ['PUBLIC_ARTIFACT_URL': 'https://dummy_link'])
+        binding.setVariable('env', ['JOB_NAME': 'dummy_integ_test'])
```
Comment (gaiksaya) [inline tests/jenkins/lib-testers/RunIntegTestScriptLibTester.groovy:34]: You can add these to same env on line 32

**Review by gaiksaya — CHANGES_REQUESTED** [review]
basepath is going null/null in all scripts. Maybe fix that? Also add tests t

**Comment by zelinh** [comment]
> basepath is going null/null in all scripts. Maybe fix that?

@gaiksaya Fixed in the latest commit. Could you help take a look again? Thanks.

**Review by gaiksaya — APPROVED** [review]
Looks good!

Bump the version in build.gradle to 4.3.0 before merging this. 
Thanks!

**Review by peterzhuamazon — APPROVED** [review]

**Comment by peterzhuamazon** [comment]
Need to bump version to 4.2.2 in build.gradle.

**Pull request merged**